src/utils.ts

```diff
@@ -28,3 +28,49 @@ export async function getCmdOutput(
   }
   return stdout;
 }
+
+export async function withRetries<T>(
+  operation: () => Promise<T>,
+  maxRetryAttempts: number,
+  isRetriable: (error: unknown) => boolean
```

Comment:
The abstraction is nice but we pass `() => true` here in both usages so perhaps keep it lean and remove that?